app/components/pages/App/index.js

```diff
@@ -33,22 +33,51 @@ export const App = () => {
 
   // save last visited path in session
   const lastVisitedPathFieldName = `${appName}${LAST_VISITED_PATH}`;
+  const historyFieldName = `${appName}${HISTORY_PATH}`;
   const setLastVisitedToSession = (path) => {
     const sanitizedPath = path.replace(prefix, '');
     // handle non-index and index paths separately
-    sessionStorage.setItem(lastVisitedPathFieldName, (sanitizedPath === INDEX_PATH ? '/' : sanitizedPath));
+    sessionStorage.setItem(lastVisitedPathFieldName, ((isEmpty(sanitizedPath) || sanitizedPath === INDEX_PATH) ? '/' : sanitizedPath));
   };
   const redirectToLastVisitedPath = () => {
-    // if the page is reloaded, redirect to the last visited page using history
+    // check type of operation done on browser
     const navType = window?.performance?.navigation?.type;
-    if (!isNil(navType) && navType === window.performance.navigation.TYPE_RELOAD) {
-      const lastPath = sessionStorage.getItem(lastVisitedPathFieldName);
+    if (!isNil(navType)) {
+      let lastPath = '';
+      // if the page is reloaded, redirect to the last visited page using last visited path session item
+      if (navType === window.performance.navigation.TYPE_RELOAD) {
+        lastPath = sessionStorage.getItem(lastVisitedPathFieldName);
+      }
+      // if the browser is navigated back and forth, redirect to the last visited page using nav history list
+      else if (navType === window.performance.navigation.TYPE_BACK_FORWARD) {
+        const str = sessionStorage.getItem(historyFieldName);
+        let existingHistory = !isEmpty(str) ? JSON.parse(str) : [];
+        // extract last path from history list
+        lastPath = existingHistory.splice(existingHistory.length - 2, 2)?.[0];
```

Comment:
(existingHistory.length - 2, 2) - this would not work when prefix is made of more than 2 parts. eg. `/unit/test/ui`

Reply:
existingHistory is made up of list of paths like ["/", "/home", "/home/contact"] and we take 2nd last from it
as last 1 is current path and 2nd last is the state we would have to go to when Back btn is clicked to navigate

changed the 2nd last path extraction from splice to calling pop() 2 times, so no dependency on length
for >= 2 len, it will give 2nd last
for < 2 len, it will give undefined and hence wont traverse (if length is 1 or 0, current path only will be there in history ideally)

Reply:
Noted.